core/blockly.js

```diff
@@ -149,6 +155,38 @@ Blockly.resizeSvgContents = function(workspace) {
   workspace.resizeContents();
 };
 
+/**
+ * Check whether the touch identifier on the event matches the current saved
+ * identifier.  If there is no identifier, that means it's a mouse event and
+ * we'll use the identifier "mouse".  This means we won't deal well with
+ * multiple mice being used at the same time.  That seems okay.
+ * If the current identifier was unset, save the identifier from the
+ * event.
+ * @param {!Event} e Mouse event or touch event.
+ * @return {boolean} Whether the identifier on the event matches the current
+ *     saved identifier.
+ */
+Blockly.checkTouchIdentifier = function(e) {
+  var identifier = (e.changedTouches && e.changedTouches.item(0) &&
+      e.changedTouches.item(0).identifier != undefined) ?
+      e.changedTouches.item(0).identifier : "mouse";
```

Comment:
Do you know if `changedTouches` will always return the touches in the same order as they appeared? E.g.:

Say the user starts a touch with identifier 'A'. Drags it around a bit. Then starts a simultaneous touch with identifier 'B'. Will changedTouches be [A, B] necessarily, or could it come back as [B, A], or some order that varies over the course of the touches?